src/Compilers/VisualBasic/Portable/Semantics/AccessCheck.vb

```diff
@@ -448,36 +400,118 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return False
         End Function
 
-        ' Determine if "derivedType" inherits from "baseType", ignoring constructed types, and dealing
-        ' only with original types.
-        Private Shared Function InheritsFromIgnoringConstruction(derivedType As TypeSymbol,
-                                                                 baseType As TypeSymbol,
-                                                                 basesBeingResolved As ConsList(Of Symbol),
-                                                                 <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo)) As Boolean
+        ''' <summary>
+        ''' Determine if "derivedType" inherits from or implements "baseType", ignoring constructed types, and dealing
+        ''' only with original types.
+        ''' </summary>
+        Private Shared Function InheritsFromOrImplementsIgnoringConstruction(derivedType As TypeSymbol,
+                                                                             baseType As TypeSymbol,
+                                                                             basesBeingResolved As BasesBeingResolved,
+                                                                             <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo)) As Boolean
             Debug.Assert(derivedType.IsDefinition)
             Debug.Assert(baseType.IsDefinition)
 
+            Dim interfacesLookedAt As PooledHashSet(Of NamedTypeSymbol) = Nothing
+            Dim baseInterfaces As ArrayBuilder(Of NamedTypeSymbol) = Nothing
+
+            Dim baseTypeIsInterface = baseType.IsInterfaceType()
+
+            If baseTypeIsInterface Then
+                interfacesLookedAt = PooledHashSet(Of NamedTypeSymbol).GetInstance()
+                baseInterfaces = ArrayBuilder(Of NamedTypeSymbol).GetInstance()
+            End If
+
             Dim current As TypeSymbol = derivedType
+            Dim result As Boolean = False
 
             While current IsNot Nothing
-                If current.Equals(baseType) Then
-                    Return True
+                If baseTypeIsInterface = current.IsInterfaceType() AndAlso current.Equals(baseType) Then
+                    result = True
+                    Exit While
+                End If
+
+                If baseTypeIsInterface Then
+                    GetBaseInterfaces(current, baseInterfaces, interfacesLookedAt, basesBeingResolved)
```

Comment:
>GetBaseInterfaces [](start = 20, length = 17)

`Get` here makes me think there should be a return that you're ignoring. Consider something like `FindBaseInterfaces`.

Reply:
Or `PopulateBaseInterfaces`. Just something that doesn't imply a return.

---
In reply to: [315436500](https://github.com/dotnet/roslyn/pull/38103#discussion_r315436500) [](ancestors = 315436500)